packages/core/src/particle/ParticleGenerator.ts

```
@@ -1025,12 +1058,133 @@ export class ParticleGenerator {
 
     this._firstFreeElement = nextFreeElement;
+
+    // ─── Sub-emitter Birth dispatch ──
+    // Skip when this very emit was triggered BY a sub-emitter (avoids self-recursion);
+    // also skip when the module has no slots at all (cheap early-out).
+    const subEmitters = this.subEmitters;
+    if (
+      !this._suppressSubEmitterDispatch &&
+      subEmitters.enabled &&
+      subEmitters.subEmitters.length > 0
+    ) {
```

ESLint

[error] Replace `⏎······!this._suppressSubEmitterDispatch·&&⏎······subEmitters.enabled·&&⏎······subEmitters.subEmitters.length·>·0⏎····` with `!this._suppressSubEmitterDispatch·&&·subEmitters.enabled·&&·subEmitters.subEmitters.length·>·0`

(prettier/prettier)